fl4health/processing/VAE_dim_reduction.py

```diff
@@ -0,0 +1,87 @@
+from abc import ABC
+from pathlib import Path
+from typing import Tuple
+
+import numpy as np
+import torch
+
+from fl4health.model_bases.autoencoders_base import ConditionalVAE, VarioationalAE
+
+
+class Processing(ABC):
+    """
+    Abstact class to add processing methods.
+    """
+
+    def __init__(
+        self,
+        checkpointing_path: Path,
+    ) -> None:
+        self.checkpointing_path = checkpointing_path
+        self.DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+
+    def load_VAE(self) -> VarioationalAE:
```

Comment:
Since you already doing the type casting for my py in the constructor of the Processor methods, I think you can have a single load AE method that returns AutoEncoderBase. Then we don't need to define a new method for each AE type we add (ex Basic AE if we end up adding that as a processor)

Reply:
I tried to do it initially, and that is why I put the load_VAE in the Processing class. It was initially a load_model() for all types of AutoEncoderBase models. But mypy doesn't like it since in AutoEncoderBase we don't have the abstract encode() function (we just have the forward()). And I can't easily put the encode function in AutoEncoderBase class since the encode function for VAE and CVAE get different inputs (ConditionalVAE.encode() also gets the condition). I can do something to make ConditionalVAE.encode() and VariationalAE.encode() function inputs have the same format (just input) but I was thinking it could make the models less readable.
Now that we have only one processor for each of the models, maybe I can move this load functions in the child classes. 
Let me know what you think, but it was pretty a good point that I was also on the fence about.